src/ra.erl

```diff
@@ -233,13 +244,20 @@ start_cluster(ClusterName, Machine, ServerIds) ->
                    ra_server:machine_conf(), [ra_server_id()]) ->
     ok | {error, term()}.
 start_server(ClusterName, ServerId, Machine, ServerIds) ->
+    start_server(ClusterName, ServerId, Machine, ?DEFAULT_SNAPSHOT_MODULE, ServerIds).
+
+-spec start_server(ra_cluster_name(), ra_server_id(),
+                   ra_server:machine_conf(), module(), [ra_server_id()]) ->
+    ok | {error, term()}.
+start_server(ClusterName, ServerId, Machine, SnapshotModule, ServerIds) ->
```

Comment:
As discussed out of bound the snapshot module should be provided as an optional callback from the `ra_machine` implementation rather than configuration. `ra_machine` and `ra_snapshot` are typically linked together and being able to configure them independently isn't ideal.